src/pyhf/optimize/autodiff_minuit.py

```diff
@@ -0,0 +1,51 @@
+"""Helper Classes for use of automatic differentiation."""
+import iminuit
+import numpy as np
+from .. import get_backend
+
+
+class AutoDiffOptimizerMixin(object):
+    """Mixin Class to build optimizers that use automatic differentiation."""
+
+    def __init__(*args, **kwargs):
+        """Create Mixin for autodiff-based optimizers."""
+
+    def minimize(
+        self,
+        objective,
+        data,
+        pdf,
+        init_pars,
+        par_bounds,
+        fixed_vals=None,
+        return_fitted_val=False,
+        return_uncertainties=False,
+    ):
+        """
+        Find Function Parameters that minimize the Objective.
+
+        Returns:
+            bestfit parameters
+
+        """
+        tensorlib, _ = get_backend()
+        tv, fixed_values_tensor, func_and_grad, init, bounds = self.setup_minimize(
+            objective, data, pdf, init_pars, par_bounds, fixed_vals
+        )
+        func = lambda pars: func_and_grad(pars)[0]
+        grad = lambda pars: func_and_grad(pars)[1]
+
+        fitresult = iminuit.minimize(
+            func, init, method='SLSQP', jac=grad, bounds=bounds
```

Comment:
the method kwarg doesn't make any sense here (maybe it's just ignored by iminuit)

Reply:
no worries. the entire file is gone now :)